File: src/main/java/io/vertx/ext/asyncsql/ConnectionCommands.java
package io.vertx.ext.asyncsql;

import io.vertx.codegen.annotations.ProxyClose;
import io.vertx.codegen.annotations.VertxGen;
import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;

/**
 * @author <a href="http://www.campudus.com">Joern Bernhardt</a>.
 */
@VertxGen(concrete = false)
public interface ConnectionCommands {

  /**
   * Starts a transaction on this connection.
   *
   * @param handler Tells the caller whether the transaction command was successful.
   */
  void startTransaction(Handler<AsyncResult<Void>> handler);

  /**
   * Commits a transaction.
   *
   * @param resultHandler Callback if commit succeeded.
   */
  void commit(Handler<AsyncResult<Void>> resultHandler);

  /**
   * Rolls back a transaction.
   *
   * @param resultHandler Callback if rollback succeeded.
   */
  void rollback(Handler<AsyncResult<Void>> resultHandler);

  /**
   * Frees the connection and puts it back into the pool.
   *
   * @param handler Callback to show whether it succeeded.
   */
  @ProxyClose
  void close(Handler<AsyncResult<Void>> handler);

}

File: src/main/resources/vertx-mysql-postgresql-js/connection_commands.js
/*
 * Copyright 2014 Red Hat, Inc.
 *
 * Red Hat licenses this file to you under the Apache License, version 2.0
 * (the "License"); you may not use this file except in compliance with the
 * License.  You may obtain a copy of the License at:
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

/** @module vertx-mysql-postgresql-js/connection_commands */
var utils = require('vertx-js/util/utils');

var io = Packages.io;
var JsonObject = io.vertx.core.json.JsonObject;
var JConnectionCommands = io.vertx.ext.asyncsql.ConnectionCommands;

/**

 @class
*/
var ConnectionCommands = function(j_val) {

  var j_connectionCommands = j_val;
  var that = this;

  /**
   Starts a transaction on this connection.

   @public
   @param handler {function} 
   */
  this.startTransaction = function(handler) {
    var __args = arguments;
    if (__args.length === 1 && typeof __args[0] === 'function') {
      j_connectionCommands.startTransaction(function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
    } else utils.invalidArgs();
  };

  /**
   Commits a transaction.

   @public
   @param resultHandler {function} 
   */
  this.commit = function(resultHandler) {
    var __args = arguments;
    if (__args.length === 1 && typeof __args[0] === 'function') {
      j_connectionCommands.commit(function(ar) {
      if (ar.succeeded()) {
        resultHandler(null, null);
      } else {
        resultHandler(null, ar.cause());
      }
    });
    } else utils.invalidArgs();
  };

  /**
   Rolls back a transaction.

   @public
   @param resultHandler {function} 
   */
  this.rollback = function(resultHandler) {
    var __args = arguments;
    if (__args.length === 1 && typeof __args[0] === 'function') {
      j_connectionCommands.rollback(function(ar) {
      if (ar.succeeded()) {
        resultHandler(null, null);
      } else {
        resultHandler(null, ar.cause());
      }
    });
    } else utils.invalidArgs();
  };

  /**
   Frees the connection and puts it back into the pool.

   @public
   @param handler {function} 
   */
  this.close = function(handler) {
    var __args = arguments;
    if (__args.length === 1 && typeof __args[0] === 'function') {
      j_connectionCommands.close(function(ar) {
      if (ar.succeeded()) {
        handler(null, null);
      } else {
        handler(null, ar.cause());
      }
    });
    } else utils.invalidArgs();
  };

  // A reference to the underlying Java delegate
  // NOTE! This is an internal API and must not be used in user code.
  // If you rely on this property your code is likely to break if we change it / remove it without warning.
  this._jdel = j_connectionCommands;
};

// We export the Constructor function
module.exports = ConnectionCommands;